### PR TITLE
ZEPPELIN-3242. Listener threw an exception java.lang.NPEat o.a.zeppelin.spark.Utils.getNoteId(Utils.java:156)

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
This issue also cause spark url can not be displayed in frontend. The root cause is that PySparkInterpreter/IPySparkInterpreter doesn't set JobGroup correctly. 

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3242

### How should this be tested?
* CI pass and also manually verified. 

### Questions:
* Does the licenses files need update? NO
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
@Tagar  Could you help try this PR ? Thanks

**Comment by Tagar** [comment]
thank you @zjffdu - I will give this a try today

File: spark/interpreter/src/test/java/org/apache/zeppelin/spark/NewSparkInterpreterTest.java [hunk 1]
```diff
@@ -385,5 +396,7 @@ public void onUpdate(int index, InterpreterResultMessageOutput out) {
               }
             })
     );
+    context.setClient(mockRemoteEventClient);
```
Comment (Tagar) [inline spark/interpreter/src/test/java/org/apache/zeppelin/spark/NewSparkInterpreterTest.java]: Is this a good idea to have a similar test for OldSparkInterpreter too?

Comment (zjffdu) [reply spark/interpreter/src/test/java/org/apache/zeppelin/spark/NewSparkInterpreterTest.java]: OldSparkInterpreter already has this kind of testing (but doesn't use mock). https://github.com/apache/zeppelin/blob/master/spark/interpreter/src/test/java/org/apache/zeppelin/spark/OldSparkInterpreterTest.java#L340

**Comment by zjffdu** [comment]
Thanks for testing @Tagar , will merge it soon

File: spark/interpreter/src/main/java/org/apache/zeppelin/spark/IPySparkInterpreter.java [hunk 2]
```diff
@@ -98,6 +99,16 @@ public BaseZeppelinContext buildZeppelinContext() {
...
   }
 
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context) {
+    InterpreterContext.set(context);
+    sparkInterpreter.populateSparkWebUrl(context);
+    String jobGroupId = Utils.buildJobGroupId(context);
+    String jobDesc = "Started by: " + Utils.getUserName(context.getAuthenticationInfo());
+    String setJobGroupStmt = "sc.setJobGroup('" +  jobGroupId + "', '" + jobDesc + "')";
+    return super.interpret(setJobGroupStmt +"\n" + st, context);
```
Comment (Tagar) [inline spark/interpreter/src/main/java/org/apache/zeppelin/spark/IPySparkInterpreter.java]: @zjffdu This brought in a minor side effect described in https://issues.apache.org/jira/browse/ZEPPELIN-3484

**Pull request closed without merging**